patches/minecraft/net/minecraft/world/level/block/JukeboxBlock.java.patch

```diff
@@ -0,0 +1,21 @@
+--- a/net/minecraft/world/level/block/JukeboxBlock.java
++++ b/net/minecraft/world/level/block/JukeboxBlock.java
+@@ -27,7 +_,7 @@
+ import net.minecraft.world.level.gameevent.GameEvent;
+ import net.minecraft.world.phys.BlockHitResult;
+ 
+-public class JukeboxBlock extends BaseEntityBlock {
++public class JukeboxBlock extends BaseEntityBlock implements net.minecraftforge.common.extensions.IForgeJukebox {
+    public static final BooleanProperty f_54254_ = BlockStateProperties.f_61439_;
+ 
+    public JukeboxBlock(BlockBehaviour.Properties p_54257_) {
+@@ -57,7 +_,8 @@
+       }
+    }
+ 
+-   public void m_238345_(@Nullable Entity p_238346_, LevelAccessor p_238347_, BlockPos p_238348_, BlockState p_238349_, ItemStack p_238350_) {
```

Comment:
This is a binary breaking change. You need to keep the original method and redirect it to your new method.

Reply:
Is this the reason for many of the self() private methods I saw in some of the extensions?

Reply:
:woman_shrugging: I just know that removing vanilla methods breaks anybody who may be using them.

Reply:
Ah. I see the confusion. Might be good to read the full patch. I just add a comment to use IForgeJukebox instead.

Reply:
This is incorrect, you are renaming the method to `setRecord`, you should instead add a different method that delegates

Reply:
Changed to insertRecord. Look good?

Reply:
LGTM

Reply:
Implemented insertRecord directly due to a suggestion from forge discrd's #github-discussions.